phantazm-server/src/main/java/com/github/phantazmnetwork/server/config/loader/ServerConfigProcessor.java

```diff
@@ -79,21 +78,21 @@ public ServerConfigProcessor(@NotNull MiniMessage miniMessage) {
     }
 
     @Override
-    public @NotNull ConfigElement elementFromData(@NotNull ServerConfig serverConfig) {
-        ConfigNode serverInfo = new LinkedConfigNode();
+    public @NotNull ConfigElement elementFromData(@NotNull ServerConfig serverConfig) throws ConfigProcessException {
+        ConfigNode serverInfo = new LinkedConfigNode(5);
         ServerInfoConfig serverInfoConfig = serverConfig.serverInfoConfig();
-        serverInfo.put("serverIP", new ConfigPrimitive(serverInfoConfig.serverIP()));
-        serverInfo.put("port", new ConfigPrimitive(serverInfoConfig.port()));
-        serverInfo.put("optifineEnabled", new ConfigPrimitive(serverInfoConfig.optifineEnabled()));
-        serverInfo.put("authType", new ConfigPrimitive(serverInfoConfig.authType().name()));
-        serverInfo.put("velocitySecret", new ConfigPrimitive(serverInfoConfig.velocitySecret()));
+        serverInfo.putString("serverIP", serverInfoConfig.serverIP());
+        serverInfo.putNumber("port", serverInfoConfig.port());
+        serverInfo.putBoolean("optifineEnabled", serverInfoConfig.optifineEnabled());
+        serverInfo.putString("authType", serverInfoConfig.authType().name());
+        serverInfo.putString("velocitySecret", serverInfoConfig.velocitySecret());
 
-        ConfigNode pingList = new LinkedConfigNode();
+        ConfigNode pingList = new LinkedConfigNode(1);
         PingListConfig pingListConfig = serverConfig.pingListConfig();
-        String description = miniMessage.serialize(pingListConfig.description());
-        pingList.put("description", new ConfigPrimitive(description));
+        pingList.put("description", AdventureConfigProcessors.component().elementFromData(pingListConfig
```

Comment:
if you want to do it in this PR, there's another miniMessage deserialization above